subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/BuildVerdictPlugin.kt

```diff
@@ -0,0 +1,36 @@
+package com.avito.android.build_verdict
+
+import com.avito.kotlin.dsl.isRoot
+import com.avito.utils.logging.ciLogger
+import com.google.gson.GsonBuilder
+import org.gradle.api.Plugin
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.kotlin.dsl.create
+
+internal data class TaskPath(val value: String)
+
+class BuildVerdictPlugin : Plugin<ProjectInternal> {
+
+    override fun apply(target: ProjectInternal) {
+        require(target.isRoot()) {
+            "build-verdict plugin must be applied to the root project"
+        }
+        val logs = mutableMapOf<TaskPath, StringBuilder>()
+        val extension = target.extensions.create<BuildVerdictPluginExtension>("buildVerdict")
+        target.gradle.addListener(TaskErrorOutputCaptureExecutionListener(logs))
```

Comment:
It may cause a memory leak because of a daemon.
Probably, we should remove this listener.
See example in `GradleCollector.cleanup`.

Also, I know an option to reveal this problem.
Run multiple builds with a daemon. Collect memory profile and check multiple instances of listeners.
It can be done easier by gradle-profiler:
```
configuration {
    tasks = ["<any simple task>"]
    // To reveal memory-leaks
    warm-ups = 10
}
```

Reply:
Also, there is an unobvious behavior of `buildFinished`. It's invoked multiple times with different results.
We had a memory leak after IDE synchronization because of it (MBS-5738).

Reply:
This is why I think I'd good to adapt and reuse existing abstractions from the `gradle-profile` module.

Reply:
I won't check if this is true. But I will add `removeListener` for both: taskOutputListner and buildFinishedAction
@eugene-krivobokov Take a look at the `buildFinishedAction` it seems to behave not like the same listener, but I didn't check